test/v3/controllers/validators/CreatePeriodSummaryValidatorFactorySpec.scala

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v3.controllers.validators
+
+import api.models.domain.{BusinessId, Nino}
+import api.models.errors._
+import api.models.utils.JsonErrorValidators
+import play.api.libs.json._
+import support.UnitSpec
+import v3.models.request.createPeriodSummary._
+
+class CreatePeriodSummaryValidatorFactorySpec extends UnitSpec with JsonErrorValidators {
+
+  private implicit val correlationId: String = "1234"
+
+  private val validNino       = "AA123456A"
+  private val validBusinessId = "XAIS12345678901"
+
+  private val validPeriodDates = Json.parse("""
+      |{
+      |  "periodStartDate": "2019-08-24",
+      |  "periodEndDate": "2020-08-24"
```

Comment:
Previous version used the fixture that had equal dates (see [here](https://github.com/hmrc/self-employment-business-api/blob/e4622d4ee989227994461f15e333ccd4a02a31b6/test/v1/fixtures/CreatePeriodSummaryFixture.scala#L28-L29)), if these are changed to be equal, you get a RULE_END_DATE_BEFORE_START_DATE which I don't think you did before. All because ResolveDateRange does not allow equal dates but the old validator did.

Reply:
@jeremystone yeah. I saw your message to Fatih so I can amend based on the outcome